pkg/controllers/common/cnstore.go

```diff
@@ -151,10 +156,24 @@ type StoreScore struct {
 	PipelineCount int `json:"pipelineCount"`
 
 	StartedTime *time.Time `json:"startedTime,omitempty"`
+	Phase       string     `json:"phase,omitempty"`
 }
 
 func (s *StoreScore) GenDeletionCost() int {
-	return s.SessionCount
+	deletionCost := s.SessionCount*sessionWeight + s.PipelineCount*pipelineWeight
```

Comment:
sessionCount and pipelineCount can go easily over phaseWeight, e.g. serving 1000 sessions in one host is a common case